tree/ntuple/v7/src/RFieldMeta.cxx

```diff
@@ -149,24 +149,55 @@ void ROOT::Experimental::RClassField::Attach(std::unique_ptr<RFieldBase> child,
    RFieldBase::Attach(std::move(child));
 }
 
-void ROOT::Experimental::RClassField::AddReadCallbacksFromIORules(const std::span<const ROOT::TSchemaRule *> rules,
-                                                                  TClass *classp)
+std::vector<const ROOT::TSchemaRule *> ROOT::Experimental::RClassField::FindRules(const RFieldDescriptor *fieldDesc)
 {
-   for (const auto rule : rules) {
+   ROOT::Detail::TSchemaRuleSet::TMatches rules;
+   const auto ruleset = fClass->GetSchemaRules();
+   if (!ruleset)
+      return rules;
+
+   if (!fieldDesc) {
+      // If we have no on-disk information for the field, we still process the rules on the current in-memory version
+      // of the class
+      rules = ruleset->FindRules(fClass->GetName(), fClass->GetClassVersion(), fClass->GetCheckSum());
+   } else {
+      // We do have an on-disk field that correspond to the current RClassField instance. Ask for rules matching the
+      // on-disk version of the field.
+      if (fieldDesc->GetTypeChecksum()) {
+         rules = ruleset->FindRules(fieldDesc->GetTypeName(), fieldDesc->GetTypeVersion(),
+                                    *fieldDesc->GetTypeChecksum());
+      } else {
+         rules = ruleset->FindRules(fieldDesc->GetTypeName(), fieldDesc->GetTypeVersion());
+      }
+   }
+
+   // For the time being, we only support rules targeting transient members
+   auto referencesNonTransientMembers = [classp = fClass](const ROOT::TSchemaRule *rule) {
+      if (rule->GetTarget() == nullptr)
```

Comment:
You might probably be able to also apply those rules as they implicitly applies to the whole object and thus need to be 'scheduled' after all other I/O as been performed.

Reply:
A `return false` here means that the rule is currently applied, cf `std::remove_if`

Reply:
Rearranged that code block and ensured that whole-object rules are executed last.